addons/pos_loyalty/models/pos_session.py

```diff
@@ -43,7 +43,7 @@ def _loader_params_loyalty_reward(self):
                 'fields': ['description', 'program_id', 'reward_type', 'required_points', 'clear_wallet', 'currency_id',
                     'discount', 'discount_mode', 'discount_applicability', 'all_discount_product_ids', 'is_global_discount',
                     'discount_max_amount', 'discount_line_product_id',
-                    'multi_product', 'reward_product_ids', 'reward_product_qty', 'reward_product_uom_id'],
+                    'multi_product', 'reward_product_ids', 'reward_product_qty', 'reward_product_uom_id', 'reward_product_domain'],
```

Comment:
Can't you simply remove `all_discount_product_ids` from the fields and only fetch the `reward_product_domain`, which would be computed in a separate compute method ?

Reply:
Hello, this is going to be the case for master, but in order to keep compatibility with the old stable behaviour, we don't really have the choice to do it like this.

Reply:
@Feyensv This problem is not actually pos-specific: any time a reward is applied, if it can apply to all products, the field is computed and will contain the product ids of all products. This leads to terrible performance on DBs with lots of products (think ~200k) in all flows that involve rewards.

We can't remove the reward_product_ids from the loaded data in stable because existing JS customizations might crash if they rely on it being present. It can definitely be a separate compute method though. I've asked david to split the compute method, there's no good reason to share the compute between the fields anyway.

If you think the potential performance improvement in other flows are not worth it, let us know and we can instead revert the changes outside of pos_loyalty, and instead of loading the field normally, load it conditionally with an override of `load_pos_data` to avoid the performance hit specifically in the case of the pos loading.

Reply:
@davidmonnom let's rephrase the commit message to explain that this doesn't only impact PoS loading